oauth2_http/java/com/google/auth/oauth2/ServiceAccountCredentials.java

```diff
@@ -935,8 +948,8 @@ public Map<String, List<String>> getRequestMetadata(URI uri) throws IOException
     // Otherwise, use self signed JWT with uri as the audience.
     JwtCredentials jwtCredentials;
     if (!createScopedRequired() && useJwtAccessWithScope) {
-      // Create JWT credentials with the scopes.
-      jwtCredentials = createSelfSignedJwtCredentials(null);
+      // Reuse selfSignedJwtCredentialsWithScope to improve the performance.
+      jwtCredentials = selfSignedJwtCredentialsWithScope;
```

Comment:
If the `selfSignedJwtCredentialsWithScope` expired (credential created 60m+ ago), the API call with the JWT will fail. How does the user trigger recreating a new JWT? Do they have to create a new credential? If that the case, we might want to check `exp` of the JWT here, and create a new one and store if the JWT expired or close to expiration.

Reply:
The `selfSignedJwtCredentialsWithScope` is a JwtCredential object. At line 957 when its `getRequestMetadata`  is called, and it refreshes its token automatically:

https://github.com/googleapis/google-auth-library-java/blob/main/oauth2_http/java/com/google/auth/oauth2/JwtCredentials.java#L161-L164